mbed_flasher/common.py

```diff
@@ -131,11 +131,14 @@ def retry(logger, func, func_args, retries=DEFAULT_RETRY_AMOUNT, conditions=None
             if error.return_code not in conditions:
                 raise error
 
-        index_from_one = index + 1
-        retry_interval = index_from_one ** 2
-        logger.info("Starting retry round {} after {} sleep"
-                    .format(index_from_one, retry_interval))
-        time.sleep(retry_interval)
+            index_from_one = index + 1
+            if index_from_one == retries:
+                raise error
```

Comment:
this makes sense but looks that for loop now returns or raises every time - Is there anymore any path which goes to lines after for loop: https://github.com/ARMmbed/mbed-flasher/blob/e18de65e2ca7d7c7c554f524c9522fedde6f5e5c/mbed_flasher/common.py#L143-L147

Reply:
Those lines can be removed now.